var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -96,6 +96,7 @@ class Esmf(MakefilePackage):
     # https://earthsystemmodeling.org/docs/release/latest/ESMF_usrdoc/node10.html#SECTION000105000000000000000
     variant("esmf_comm", default="auto", description="Override for ESMF_COMM variable")
     variant("esmf_os", default="auto", description="Override for ESMF_OS variable")
+    variant("beta", default=False, description="Dummy variant for beta versions")
```

Comment:
What if we called this variant `b09` with description `beta snapshot09`? Sure, we may have to add a few more beta snapshots here, but I don't think we want to post these changes in the spack authoritative repo, since anything pushed out to the larger spack community should be using real releases, not beta snapshots. But this way, we can control the module name explicitly for every `bXY` that we are adding.

Reply:
FYI, pretty soon MAPL will be using ESMF v8.5.0b13

We don't mean to make your life hard, but, well...

Reply:
Note: Technically not *required* yet, but ESMF added some new functionality we hope to start using. I'll be holding off requiring 8.5.0 until it is truly required (GEOS will move to it because of Baselibs).

Reply:
I like the named variants approach. It allows the user to try different beta snapshots with one control. I think this snapshot mechanism would be useful for netcdf libraries (and probably many more) since Unidata provides beta snapshots too (they call them "release candidates").

Reply:
No matter what happens, I'm guessing the spack folks will have thoughts when it gets to them. @climbfuji knows what happened when I tried to make what I thought was a simple change to FMS. 

I learned a lot in that PR. 😄

Reply:
> I like the named variants approach. It allows the user to try different beta snapshots with one control, allows two or more snapshots to co-exist, and doesn't put actual beta shapshot versions in the authoritative repo. I think this snapshot mechanism would be useful for netcdf libraries (and probably many more) since Unidata provides beta snapshots too (they call them "release candidates").

I'm good with that. If it sounds good to all, I'm thinking I'll have a named variant with no pre-specified values, then it's just a matter of making sure to add to modules.yaml as we go.